dev/RadioButtons/ColumnMajorUniformToLargestGridLayout.h

```diff
@@ -0,0 +1,48 @@
+﻿#pragma once
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+#include "common.h"
+#include "ColumnMajorUniformToLargestGridLayout.g.h"
+#include "ColumnMajorUniformToLargestGridLayout.properties.h"
+#include "Layout.h"
+#include "NonVirtualizingLayout.h"
+
+class ColumnMajorUniformToLargestGridLayout :
+    public ReferenceTracker<ColumnMajorUniformToLargestGridLayout, winrt::implementation::ColumnMajorUniformToLargestGridLayoutT, NonVirtualizingLayout>,
+    public ColumnMajorUniformToLargestGridLayoutProperties
+{
+public:
+    ColumnMajorUniformToLargestGridLayout();
+    winrt::Size MeasureOverride(
+        winrt::NonVirtualizingLayoutContext const& context,
+        winrt::Size const& availableSize);
+    winrt::Size ArrangeOverride(
+        winrt::NonVirtualizingLayoutContext const& context,
+        winrt::Size const& finalSize);
+
+    void OnColumnSpacingPropertyChanged(const winrt::DependencyPropertyChangedEventArgs&);
+    void OnRowSpacingPropertyChanged(const winrt::DependencyPropertyChangedEventArgs&);
+    void OnMaximumColumnsPropertyChanged(const winrt::DependencyPropertyChangedEventArgs&);
+
+    //Testhooks helpers, only function while m_testHooksEnabled == true
+    void SetTestHooksEnabled(bool enabled);
+    int GetRows();
+    int GetColumns();
+    int GetLargerColumns();
+    winrt::event_token LayoutChanged(winrt::TypedEventHandler<winrt::ColumnMajorUniformToLargestGridLayout, winrt::IInspectable> const& value);
+    void LayoutChanged(winrt::event_token const& token);
+
+private:
+    winrt::Size LargestChildSize(const winrt::NonVirtualizingLayoutContext& context);
+
+    //Testhooks helpers, only function while m_testHooksEnabled == true
+    bool m_testHooksEnabled{ false };
+    winrt::event<winrt::TypedEventHandler<winrt::ColumnMajorUniformToLargestGridLayout, winrt::IInspectable>> m_layoutChangedEventSource;
+
+    int m_rows{ -1 };
```

Comment:
Nit: All these member variables seem to be useful only in ArrangeOverride and could possibly be made stack variables. Are these here in order to test using test hooks ? Could you instead use LayoutUpdated event on UIElement and LayoutInformation.GetLayoutSlot api to get the rects and validate based on those ?

Reply:
these were promoted to member variables so I could ask for them in the tests, via the test hooks.  It is probably  possible to reconstruct these values from GetLayoutSlot...

---
In reply to: [347415956](https://github.com/microsoft/microsoft-ui-xaml/pull/1623#discussion_r347415956) [](ancestors = 347415956)